implementation/src/main/java/io/smallrye/openapi/runtime/scanner/AnnotationScannerExtension.java

```diff
@@ -0,0 +1,66 @@
+package io.smallrye.openapi.runtime.scanner;
+
+import java.util.Collection;
+
+import org.eclipse.microprofile.openapi.models.parameters.Parameter.In;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.MethodInfo;
+import org.jboss.jandex.MethodParameterInfo;
+import org.jboss.jandex.Type;
+
+import io.smallrye.openapi.runtime.util.JandexUtil.JaxRsParameterInfo;
+
+/**
+ * Extension point for supporting extensions to JAX-RS.
+ */
+public interface AnnotationScannerExtension {
+
+    /**
+     * Determines where an @Parameter can be found (examples include Query,
+     * Path, Header, Cookie, etc).
+     * 
+     * @param paramInfo
+     * @return the parameter location, or null if unknown
+     */
+    public default In parameterIn(MethodParameterInfo paramInfo) {
+        return null;
```

Comment:
I might be missing something, but why are you providing default/null implementations in this interface instead of just a non-default definition and requiring the user to provide a null impl if they really want that.

This might relate to your comment https://github.com/smallrye/smallrye-open-api/pull/50#issuecomment-450897205 in a way I'm not grokking.

Reply:
Because providing a default impl makes it instantly obvious to users what a valid default return value is. This shows that returning `null` is acceptable and will allow other extensions to participate. Otherwise you have to rely on javadoc and have no code to prove it.

Reply:
Okay, thanks, I wasn't sure whether an implementor should generally be expected to provide an implementation to all these methods for it to be sensible.